core/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -3570,6 +3591,72 @@ public long generateSeqNo() {
         }
     }
 
+    public void testMinGenerationForSeqNo() throws IOException, BrokenBarrierException, InterruptedException {
```

Comment:
> I do not see what can be added that is not already covered by other tests. I

Correct me if i missed it but I don't see any test that creates a situation in which we recovery from a translog that consists of multiple generations where the seq# are scattered across them. `testTranslogReplay` comes the closest but I think it needs some beefing up (it always writes in order, meaning that it always recovers from the last generation).